core/src/edu/cornell/gdiac/temporary/entity/Note.java

```diff
@@ -132,43 +145,119 @@ public void setDestroyed(boolean d){
 	/**
 	 * Initialize shell with trivial starting position.
 	 */
-	public Note(int line, NoteType n, int frame, Texture t) {
+	public Note(int line, NoteType n, long startSample, Texture t) {
 		// Set minimum Y velocity for this shell
 		this.line = line;
 		hitStatus = 0;
 		animeframe = 0.0f;
+		curEndFrame = 0.0f;
+		curTrailFrame = 0.0f;
 		nt = n;
-		vy = n == NoteType.HELD? 0f : descentSpeed;
-		startFrame = frame;
+		this.startSample = startSample;
+		//Set the number of animation frames
+		switch(nt) {
+			case BEAT:
+				NUM_ANIM_FRAMES = 1;
+				break;
+			case HELD:
+				NUM_ANIM_FRAMES = 1;
+				break;
+			case SWITCH:
+				NUM_ANIM_FRAMES = 1;
+				break;
+		}
 		animator = new FilmStrip(t,1,NUM_ANIM_FRAMES,NUM_ANIM_FRAMES);
 		origin = new Vector2(animator.getRegionWidth()/2.0f, animator.getRegionHeight()/2.0f);
 		h = animator.getRegionHeight();
 		w = animator.getRegionWidth();
 	}
 
+	/**
+	 * Sets the note texture
+	 * @param texture
+	 */
 	public void setTexture(Texture texture) {
 		animator = new FilmStrip(texture,1,NUM_ANIM_FRAMES,NUM_ANIM_FRAMES);
 		origin = new Vector2(animator.getRegionWidth()/2.0f, animator.getRegionHeight()/2.0f);
 		h = animator.getRegionHeight();
 		w = animator.getRegionWidth();
 	}
 
+	private FilmStrip trailAnimator;
+	private int trailFrames;
+	/**
+	 * Height of the trail texture
+	 */
+	private float trailHeight;
+	/**
+	 * Width of the trail texture
+	 */
+	private float trailWidth;
+	/**
+	 * Origin of trail texture
+	 */
+	private Vector2 trailOrigin;
 
-	public void update(int frame) {
 
-		if(nt == NoteType.HELD){
-			by += descentSpeed;
-			if(frame == (startFrame + holdFrame)){
-				vy = descentSpeed;
-			}
+	private FilmStrip endAnimator;
+	private int endFrames;
+	/**
+	 * Height of end texture
+	 */
+	private float endHeight;
+	/**
+	 * Width of end texture
+	 */
+	private float endWidth;
+	/**
+	 * Origin of end texture
+	 */
+	private Vector2 endOrigin;
+	private float curTrailFrame;
+	private float curEndFrame;
 
-		}
-		y += vy;
+	/**
+	 * Set the textures for held notes. This is outside the constructor because not every note is a held note
+	 * @param trail
+	 * @param trailFrames
+	 * @param end
+	 * @param endFrames
+	 */
+	public void setHoldTextures(Texture trail, int trailFrames, Texture end, int endFrames){
```

Comment:
why don't you put this in the constructor? I think to solve the problem you bring up, you can just call it with an `if` statement or include it in the switch statement I feel like. that way if you initialize another held note somewhere you don't have to worry about this.

Reply:
This is how they set textures in the lab. I think it's because you need to wait until populate() is ran to actually get access to the textures. And that only happens after initialization of the notes